renamed all classes monster in txt and replaced in discard

Keicabba opened the pull request:
Created class MonsterEntity, MonsterRepository, MonsterService. Populated MonsterEntity
MonsterRepository and MonsterService actually are empty

Keicabba committed:
renamed all classes monster in txt and replaced in discard
Created class MonsterEntity, MonsterRepository, MonsterService.
Populated MonsterEntity
MonsterRepository and MonsterService actually are empty

TTranchese commented on src/main/java/it/project1/monster/MonsterEntity.java:
```diff
@@ -0,0 +1,71 @@
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+
```
fai attenzione che qui manca l'annotation per entity

TTranchese reviewed (commented):
aggiungere @Entity alla MonsterEntity

Keicabba committed:
Added @Entity adn  tablename to MonsterEntity
MonsterRepository interface modified
Added Methods to MonsterService
Created MonsterConstroller

TTranchese commented on src/main/java/it/project1/monster/MonsterController.java:
```diff
@@ -0,0 +1,40 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("monsters")
```
attenzione qui che ti manca il "/" di fronte a monster, non so altrimenti poi come funziona, o se funziona lo stesso

TTranchese commented on src/main/java/it/project1/monster/MonsterController.java:
```diff
@@ -0,0 +1,40 @@
+
+    @PutMapping("/{id}")
+    public MonsterEntity updateMonster(@PathVariable Long id, @RequestBody MonsterEntity monster) {
+        monster.setMonsterId(Math.toIntExact(id));
```
a cosa ti serve il Math.toIntExact() se a .setMonsterId gli fai prendere un int?

TTranchese commented on src/main/java/it/project1/monster/MonsterEntity.java:
```diff
@@ -1,21 +1,21 @@
     @Id
-    @GeneratedValue(strategy = GenerationType.AUTO)
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "monster_id")
```
credo che per integrità, dato che anche noi abbiamo fatto così, quando crei una entity, ti convenga dargli come attributo primary key soltanto il nome "id", in questo caso magari a chi serve l'id di un mostro, sa che può scrivere "monsters.id" e trovarlo, come nel caso se ti serva un account o un item puoi scrivere tranquillamente "accounts.id" o "items.id", credo sia solo una cosa di convenzione

TTranchese commented on src/main/java/it/project1/monster/MonsterRepository.java:
```diff
@@ -1,4 +1,15 @@
+    Optional<MonsterEntity> findById(Long id);
+    MonsterEntity save(MonsterEntity monster);
+    void delete(MonsterEntity monster);
 }
```
questi a cosa servono? chiedo perché io ho scritto soltanto l'interfaccia così com'era e non ho aggiunto altro

TTranchese commented:
allora non so il perché ma mi da sempre conflitti, in generale il codice sembra ben scritto, farei solo qualche accorgimento e se magari mi spieghi qui quelle due cose che ho chiesto mi faresti un piacere, poi semmai domani vediamo di risolvere anche i conflitti post 10 minute meet

TTranchese committed:
Merge branch 'master' into monster
